tests/gtests/internals/test_sdpa.cpp

```diff
@@ -0,0 +1,1916 @@
+            }
+        }
+        return;
+    }
+#endif
+#if DNNL_GPU_RUNTIME == DNNL_RUNTIME_OCL
+    dnnl::stream s(eng);
+    if (eng.get_kind() == dnnl::engine::kind::gpu) {
+        if (mem.get_desc().get_data_type() != dnnl_f32
+                && std::is_same<T, float>::value) {
+            memory mem_f32_mem(
+                    {mem.get_desc().get_dims(), memory::data_type::f32,
+                            mem.get_desc().get_strides()},
+                    eng);
+            write_to_dnnl_memory<float>((const float *)handle, mem_f32_mem);
+            dnnl::reorder(mem_f32_mem, mem).execute(s, mem_f32_mem, mem);
+            s.wait();
+        } else if (mem.get_desc().get_data_type() != dnnl_s32
+                && std::is_same<T, int>::value) {
+            memory mem_s32_mem(
+                    {mem.get_desc().get_dims(), memory::data_type::s32,
+                            mem.get_desc().get_strides()},
+                    eng);
+            write_to_dnnl_memory<int>((const int *)handle, mem_s32_mem);
+            dnnl::reorder(mem_s32_mem, mem).execute(s, mem_s32_mem, mem);
+            s.wait();
+        } else if ((mem.get_desc().get_data_type() == dnnl_u8
+                           || mem.get_desc().get_data_type() == dnnl_s8
+                           || mem.get_desc().get_data_type() == dnnl_s4
+                           || mem.get_desc().get_data_type() == dnnl_u4)
+                && std::is_same<T, unsigned>::value) {
+            memory mem_u32_mem(
+                    {mem.get_desc().get_dims(), memory::data_type::s32,
+                            mem.get_desc().get_strides()},
+                    eng);
+            write_to_dnnl_memory<unsigned>(
+                    (const unsigned *)handle, mem_u32_mem);
+            dnnl::reorder(mem_u32_mem, mem).execute(s, mem_u32_mem, mem);
+            s.wait();
+        } else if ((mem.get_desc().get_data_type() == dnnl_f32
+                           && std::is_same<T, float>::value)
+                || (mem.get_desc().get_data_type() == dnnl_s32
+                        && std::is_same<T, int>::value)) {
+            void *mapped_ptr = mem.map_data();
+            if (mapped_ptr) std::memcpy(mapped_ptr, handle, size);
+            mem.unmap_data(mapped_ptr);
+        } else {
+            // PC: this branch is identical to the one above
+            void *mapped_ptr = mem.map_data();
+            if (mapped_ptr) std::memcpy(mapped_ptr, handle, size);
+            mem.unmap_data(mapped_ptr);
+        }
+        return;
+    }
+#endif
+
+    if (eng.get_kind() == dnnl::engine::kind::cpu) {
+        uint8_t *dst = static_cast<uint8_t *>(mem.get_data_handle());
+        if (!dst) throw std::runtime_error("get_data_handle returned nullptr.");
+        for (size_t i = 0; i < size; ++i)
+            dst[i] = ((uint8_t *)handle)[i];
+        return;
+    }
+
+    assert(!"not expected");
+}
+
+static std::random_device rd;
+static std::mt19937 generator(rd());
```

Comment:
Let's not put std objects into a global space. Let's either put a copy for each of those where they used, or make a function with a static object inside and return a reference to it.